simapp/simd/cmd/root.go

```diff
@@ -38,7 +38,7 @@ import (
 // NewRootCmd creates a new root command for simd. It is called once in the
 // main function.
 func NewRootCmd() (*cobra.Command, params.EncodingConfig) {
-	encodingConfig := simapp.MakeEncodingConfig()
+	encodingConfig := simapp.MakeTestEncodingConfig()
```

Comment:
This line is hte line being referenced in the documentation at `server/README.md`. I know that "simapp" in generally is mainly for testing purposes, but it seems strange to call this `MakeTestEncodingConfig` here, when from what I understand users will mostly be copying a lot of `simapp/simd` code into their own application when building their own chains.

Perhaps its fine if we have a better strategy for users than looking at `simapp/` code when bootstrapping their own app.

@aaronc when you said that we shouldint depend on `EncodingConfig` at all in the `server/README.md`, what do you suggest we do here instead? explicitly do all the interface registry stuff here in this function to make it super clear ?

Reply:
`MakeTestEncodingConfig` is just the wrong name for this function I'd say. What was the rationale in the first place?

Reply:
It's just for switching between amino and protobuf builds.

I don't think simapp is intended to be a reference for apps. The tutorial is for that. Simd was only added during stargate work for testing.

If we want to make simapp a reference for apps we should pull out all the legacy stuff and fully deprecate amino support which will allow EncodingConfig to go away.

Reply:
It makes sense to me, thanks. Yet IMHO the `MakeTestEncodingConfig` name does not suggest anything like that and `MakeEncodingConfig` seems still the best name?

Reply:
@alessio  - please check the referenced issue: #7642

I'm open for a better name suggestions. As stated in this PR, it's a temporal solution to make it clear for the users that they shouldn't use this method.